### ISSUE TITLE
Implement PartialEq for Value to compare with native data types Value represents

### ISSUE THREAD

**Issue by imp** [serde-rs/json#219]
For example for the following to work
```rust
        let n = Value::from(42);
        assert_eq!(n, 42);

        let s = Value::String(String::from("hello"));
        assert_eq!(s, "hello");
        assert_eq!(s, String::from("hello"));
```

**Comment by KodrAus** [comment]
@oli-obk I guess that would be `false`? What I'm wondering is what would be the result of `Value::from(42), "42"`?

**Comment by oli-obk** [comment]
@KodrAus that's "obviously" false, too.

**Comment by imp** [comment]
The idea was to allow compare with the native types that are being representing by Value. So that
```rust
let n = Value::from(42);
assert_ne!(n, "42");
assert_eq!(n, 42);
```

All the other fancy possible conversion are out of scope.

**Comment by dtolnay** [comment]
I like this idea and I think @imp's suggested semantics make sense.

**Comment by imp** [comment]
Closed via #235
